allennlp/training/metrics/metric.py

```diff
@@ -34,6 +38,26 @@ def get_metric(
         """
         raise NotImplementedError
 
+    def get_metric_name_value_pairs(
+        self, default_name: str, reset: bool = False
```

Comment:
Maybe a default default name can be created from the underlying class name :man_shrugging: